lib/main.js

```diff
@@ -18,7 +20,11 @@ Object.assign(exports, {
     PlacesProvider.links.init();
     SearchProvider.search.init();
     options.telemetry = false;
-    app = new ActivityStreams(options);
+
+    Task.spawn(function*() {
+      options.clientID = yield ClientID.getClientID();
```

Comment:
Hey, thanks for asking about the FxA stuff :)

We had a bit of discussion about this with some of the FxA team. The suggestion would be to **not** use the fxa uid as a primary identifier for this. Create your own or use the telemetry one. 

Also do not use `identity.fxaccounts.lastSignedInUserHash` in other metrics fields, this pref is used for a specific account flow and might not always be there. If you want to include the fxa details in your metrics feilds then fetch it using this component: http://mxr.mozilla.org/mozilla-central/source/services/fxaccounts/FxAccounts.jsm#462

You can also get the device id using: http://mxr.mozilla.org/mozilla-central/source/services/fxaccounts/FxAccounts.jsm#562

Let me know if you need help with those!